packages/frontend-api/src/Model/Complaint/ComplaintItemDataApiFactory.php

```diff
@@ -35,12 +35,19 @@ public function __construct(
     public function createFromComplaintItemInput(OrderItem $orderItem, array $item): ComplaintItemData
     {
         $product = $orderItem->getProduct();
+        $productName = $orderItem->getName();
+        $catnum = $orderItem->getCatnum();
+
+        if ($product) {
+            $productName = $product->getName($this->domain->getLocale());
+            $catnum = $product->getCatnum();
+        }
```

Comment:
this is an interesting problem, which data should take precedence when creating the complaint item? :smile: Have you discussed this with the product owner? From a customer point of view, I would (probably) prefer the other way round :thinking:

Reply:
fixed [fixup! complaints b2b customers support](https://github.com/shopsys/shopsys/pull/3295/commits/585c5b09e615c10c4351fc201d7abf686fa2e022)

Reply:
I checked with the PO, and your approach is preferred. Updated.

[fixup! complaints b2b customers support](https://github.com/shopsys/shopsys/pull/3295/commits/585c5b09e615c10c4351fc201d7abf686fa2e022)